### PR TITLE
Build 'test' and 'shred' on Redox

### PR DESCRIPTION
**Pull request by ids1024**

### CODE REVIEW

File: src/test/test.rs [hunk 1]
```diff
@@ -336,60 +342,66 @@ enum PathCondition {
...
+    use std::os::unix::ffi::OsStrExt;
+    use std::fs::{self, Metadata};
+    use std::ffi::OsStr;
 
-    static S_ISUID: mode_t = 0o4000;
-    static S_ISGID: mode_t = 0o2000;
-    static S_IFSOCK: mode_t = 0o140000;
+    let path = OsStr::from_bytes(path);
+
+    static S_ISUID: u32 = 0o4000;
```
Comment (Arcterus) [inline src/test/test.rs]: Can you make these `const` instead of `static`?

File: src/test/test.rs [hunk 2]
```diff
@@ -336,60 +342,66 @@ enum PathCondition {
...
-        if uid == stat.st_uid {
-            stat.st_mode & ((p as mode_t) << 6) != 0
-        } else if gid == stat.st_gid {
-            stat.st_mode & ((p as mode_t) << 3) != 0
+
+    let perm = |metadata: Metadata, p: Permission| {
+        #[cfg(not(target_os = "redox"))]
+        let (uid, gid) = unsafe { (libc::getuid(), libc::getgid()) };
+        #[cfg(target_os = "redox")]
+        let (uid, gid) = (syscall::getuid().unwrap() as u32,
```
Comment (Arcterus) [inline src/test/test.rs]: Is it possible for either of these to fail?  I can’t really think of a scenario where it would be but I just want to make sure.

Comment (ids1024) [reply src/test/test.rs]: It shouldn't be possible; looking at the [implementation](https://github.com/redox-os/kernel/blob/master/src/syscall/privilege.rs), it should only fail if the calling process doesn't exist, which obviously shouldn't happen (or execution shouldn't continue).

@jackpot51 Perhaps these functions shouldn't return `Result`s?

Comment (Arcterus) [reply src/test/test.rs]: Okay, thanks for checking.

**Comment by Arcterus** [comment]
👍

**Pull request merged**